src/model/dataset.ts

```diff
@@ -1,15 +1,15 @@
 import type { DatasetV2 } from '@datagouv/components'
 
 import type { GenericResponse } from './api'
-import type { DatasetProperties } from './topic'
+import type { ResolvedFactor } from './topic'
 
 export interface DatasetV2Response extends GenericResponse {
   data: DatasetV2[]
 }
 
 export interface DatasetModalData {
   index?: number
-  dataset?: DatasetProperties
+  element?: ResolvedFactor
```

Comment:
Feels like either refactoring went too far or not far enough.
How will that work when adding other elements than factors?
With that naming+signature, I'd expect `element?: ResolvedFactor | ResolvedReuse`, justifying the more generic `element` name?